src/snitch_cli.cpp

```diff
@@ -401,19 +401,23 @@ void print_help(std::string_view program_name, const print_help_settings& settin
 }
 
 std::optional<cli::input> parse_arguments(int argc, const char* const argv[]) noexcept {
-    // First, parse just looking for color options so we can display console messages correctly.
-    const bool with_color = impl::parse_color_options(argc, argv);
-
-    // Now parse everything for real.
-    std::optional<cli::input> ret_args =
-        impl::parse_arguments(argc, argv, impl::expected_args, {.with_color = with_color});
+    if constexpr (SNITCH_DISABLE) {
+        return {};
+    } else {
```

Comment:
Any reason this cannot be left as it was? The least the code differs in enabled/disabled, the easiest it will be to maintain. As far as I understand, this function should simply not be called if *snitch* is disabled, so it should be harmless to leave it alone?

Reply:
Regarding backward compatibility; the `SNITCH_ENABLE` option didn't exist before, so we only need to preserve backward compatibility for the case where it is `ON`. When set to `OFF`, this is new territory, so there's no compatibility concern and we can break everything.